### PR TITLE
Add (?d) ignore prefix, allowing them to be deleted

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
### Purpose

Adds a (?d) prefix which marks the file as deletable, so that when we are deleting a directory, we can delete anything that is ignored and in our way.

Also, removes the need to specify the prefixes in a certain order.

### Testing

Users will tell me if I deleted all of their photos

### Documentation

Need some

### CODE REVIEW

File: lib/ignore/ignore.go [hunk 1]
```diff
@@ -99,16 +116,16 @@ func (m *Matcher) Parse(r io.Reader, file string) error {
 	return err
 }
 
-func (m *Matcher) Match(file string) (result bool) {
+func (m *Matcher) Match(file string) (result Result) {
 	if m == nil {
-		return false
+		return Result{false, false}
```
Comment (calmh) [inline lib/ignore/ignore.go]: I'd generally prefer to see these as `Result{include: false, deletable: false}` to be explicit about what the various booleans are, or maybe just a `Result{}` as the zero value is a non-deletable non-included non-match, or as a named constant as we return it in a couple of places. But this is just a style opinion.

File: lib/ignore/ignore.go [hunk 2]
```diff
@@ -133,17 +150,23 @@ func (m *Matcher) Match(file string) (result bool) {
 				lowercaseFile = strings.ToLower(file)
 			}
 			if pattern.match.Match(lowercaseFile) {
-				return pattern.include
+				return Result{
+					pattern.include,
+					pattern.deletable,
```
Comment (calmh) [inline lib/ignore/ignore.go]: These are fine though as it's obvious from the values what they mean :)

**Comment by calmh** [comment]
Looks awesome apart from the nit about the `false, false` stuff. Clean, simple and very useful.

**Comment by AudriusButkevicius** [comment]
Updated.

**Comment by calmh** [comment]
@st-review merge this please

lib/ignore: Implement deletable ignores using (?d) prefix (fixes #1362)

**Comment by st-review** [comment]
OK, merged as 5a98af622d20f859dae1fd890ead812cdbb775f8. Thanks, @AudriusButkevicius!

**Pull request closed without merging**